scipy/stats/mstats_basic.py

```diff
@@ -536,38 +547,79 @@ def kendalltau(x, y, use_ties=True, use_missing=False):
                 for i in range(len(ry)-1)], dtype=float)
     D = np.sum([((ry[i+1:] < ry[i])*(rx[i+1:] > rx[i])).filled(0).sum()
                 for i in range(len(ry)-1)], dtype=float)
+    xties = count_tied_groups(x)
```

Comment:
In terms of performance
Taking this outside of the `if use_ties` will increase the redundant work if use_ties is False. The `stats` version does not have the use_ties option and always computes them.

one possibility would be to compute count_tied_groups `if use_ties or method="exact"` but then the program flow would need to be changes.

i.e. the question is whether there should be a fast path for floating point numbers, where we know that we don't have ties, and that is large enough for "asymptotic", and how much difference in performance that would actually make.

Reply:
@josef-pkt Good catch, I agree that this should be improved. This goes hand in hand with the outsourcing you mentioned above since restructuring the flow will be much easier once the functionality is outsourced. I'll get to work on it as soon as possible, but for now I have implemented the more urgent issues.